hack/setup_observability.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+#This script installs the prometheus and granafa operator
+#from community operator hub. It then configures them with
+#github oauth
+
+ROOT="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"/..
+
+cd $ROOT/hack/monitoring
+
+#Namespace to which we want to install observability stack
+#default is openshift-customer-monitoring
+export MONITORING_NAMESPACE="openshift-customer-monitoring"
+
+#Github access tokens for prometheus oauth
+export PROMETHEUS_GITHUB_CLIENT_ID="<base64_encoded_github_client_id_for_prometheus>"
+export PROMETHEUS_GITHUB_CLIENT_SECRET="<base64_encoded_github_client_secret_for_prometheus>"
+export PROMETHEUS_GITHUB_COOKIE_SECRET="<base64_encoded_github_cookie_secret>"
+export PROMETHEUS_K8S_SECRET_NAME="<prometheus_secret_for_serviceaccounts>"
+
+#Install prometheus operator
+oc process -p MONITORING_NAMESPACE=$MONITORING_NAMESPACE -f install_prometheus.yaml|oc apply -f -
+#Configure prometheus with github oauth
+oc process -p MONITORING_NAMESPACE=$MONITORING_NAMESPACE -p PROMETHEUS_GITHUB_CLIENT_ID=$PROMETHEUS_GITHUB_CLIENT_ID -p PROMETHEUS_GITHUB_CLIENT_SECRET=$PROMETHEUS_GITHUB_CLIENT_SECRET -p PROMETHEUS_GITHUB_COOKIE_SECRET=$PROMETHEUS_GITHUB_COOKIE_SECRET -p PROMETHEUS_K8S_SECRET_NAME=$PROMETHEUS_K8S_SECRET_NAME -f configure_prometheus.yaml|oc apply -f -
+
+#Give prometheus service account access to scrape the operators
+oc policy add-role-to-user view system:serviceaccount:openshift-customer-monitoring:prometheus-k8s -n toolchain-host-operator
+oc policy add-role-to-user view system:serviceaccount:openshift-customer-monitoring:prometheus-k8s -n toolchain-member-operator
+
+
+#Install grafana operator
+oc process -p MONITORING_NAMESPACE=$MONITORING_NAMESPACE -f install_grafana.yaml|oc apply -f -
+
+#Github access tokens for grafana oauth
+export GRAFANA_GITHUB_CLIENT_ID="<base64_encoded_github_clientid_for_grafana>"
```

Comment:
this should be described in readme that we need to run these exports before running the script, ideally with description how to configure. Then the export should not be in this script

Reply:
We dont need to run the exports before running the scripts we need to set values there. I thought it would self clarify. Let me specify them in the README

Reply:
@Michkov is right. 
Also, the script should
1) Use a default value if an env variable is not provided, but there's a reasonable default (client ID, secret name, user ID)
2) Print an error and exit if an env variable is not provided, and there's no reasonable default (OAUTH token info, password)